src/internal_telemetry/allocations/mod.rs

```diff
@@ -31,6 +127,29 @@ pub const fn init_allocation_tracing() {}
 /// a [`tracing::Span`] to achieve this" we utilize the logical invariants provided by spans --
 /// entering, exiting, and how spans exist as a stack -- in order to handle keeping the "current
 /// allocation group" accurate across all threads.
-pub fn acquire_allocation_group_id() -> AllocationGroupId {
-    AllocationGroupId::register().expect("failed to register allocation group token")
+pub fn acquire_allocation_group_id(
+    component_id: String,
+    component_type: String,
+    component_kind: String,
+) -> AllocationGroupId {
+    // try to init the thread id's here since they are lazily initialized.
+    let _ = THREAD_ID.with(|t| t.replace(THREAD_COUNTER.fetch_add(1, Ordering::Relaxed)));
```

Comment:
I'm not sure this makes sense.

We could end up registering all of our allocation groups in the very same thread, and likely are doing exactly that, given how a running topology is built... and then we'd potentially end up having all accesses to update the allocations/deallocations land in the same bucket because they're using the default value of `0` set by the TLS initializer.

Reply:
ah we do spawn after attaching the groups. Yeah this is invalid then.